### PR TITLE
chore: plumb new navigation record generation

### PR DESCRIPTION
**Pull request by RohinBhargava**
## Short description of the changes made

* plumbs new record generation into FDR write APIs
* please review carefully, especially in `getDocsWriteV2Service`
* creates `convertDocsDefinitionToRead` to merge logic and prevent duplicated code when converting Docs DB shape to Read shape

## What was the motivation & context behind this PR?

* enable new search experience

## How has this PR been tested?

* to be tested in FDR staging

### CODE REVIEW

File: servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts [hunk 1]
```diff
@@ -330,6 +346,47 @@ async function uploadToAlgolia(
...
+        lightModeEnabled: dbDocsDefinition.config.colorsV3?.type !== "dark",
+    };
+    configSegmentTuples.map(([_, indexSegment]) => {
+        const v2Records = generateAlgoliaRecords({
+            indexSegmentId: indexSegment.id,
+            nodes: FernNavigation.utils.toRootNode(loadDocsForUrlResponse),
+            pages: FernNavigation.utils.toPages(loadDocsForUrlResponse),
+            apis: FernNavigation.utils.toApis(loadDocsForUrlResponse),
+            isFieldRecordsEnabled: false,
+        });
```
Comment (dsinghvi) [inline servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts:381]: does this need to call the convert functions you added in https://github.com/fern-api/fern-platform/pull/1760/files#diff-ae4d69c24f2260f539cd0c679a6ff14552c442b1b64b32c96f0ec426e88f5acf ?

File: servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts [hunk 2]
```diff
@@ -330,6 +346,47 @@ async function uploadToAlgolia(
...
+                    createdAt: new Date(),
+                    version: null,
+                })),
+                docsDbDefinition: dbDocsDefinition,
+                app,
+            }),
+        }),
+        lightModeEnabled: dbDocsDefinition.config.colorsV3?.type !== "dark",
+    };
+    configSegmentTuples.map(([_, indexSegment]) => {
```
Comment (dsinghvi) [inline servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts:374]: I think we only need to run this piece of logic if the CLI provides `root: optional<navigationV1.RootNode>` in the `DocsConfig`  

```yml
DocsConfig:
    properties:
      title: optional<string>
      defaultLanguage: optional<commons.ProgrammingLanguage>
      announcement: optional<commons.AnnouncementConfig>

      # navigation
      navigation:
        type: optional<NavigationConfig>
        availability: deprecated
      root: optional<navigationV1.RootNode>
```

Comment (dsinghvi) [reply servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts:374]: if we want to run it across all CLI versions then we probably dont want to append to `searchRecords` and set it to `[]` on line 342

**Review by abvthecity — APPROVED** [review]

File: servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts (hunk 1 above)
Comment (RohinBhargava) [reply servers/fdr/src/controllers/docs/v2/getDocsWriteV2Service.ts:381]: The convert functions actually piggyback in the imported package, to keep the working tree somewhat clean.

**Review by abvthecity — APPROVED** [review]

**Pull request merged**